app/utils/pages.tsx

```diff
@@ -13,58 +13,54 @@ interface IPage {
 }
 
 export function getPages(): ILeagueLink[] {
-    const appLevelDir = join(cwd(), "app");
-    const currentDirFilesList = fs.readdirSync(appLevelDir);
-    let archiveDirFilesList: string[] = [];
-    const currentBBLeague = "big-brother-26";
-    const currentSurvivorLeague = "survivor-47";
-    const pages: ILeagueLink[] = [
-        {
-            name: "Current (Survivor)",
-            subpages: [{
-                name: "Contestants",
-                path: "/active/" + currentSurvivorLeague + "/contestants"
-            }]
-        },
-        {
-            name: "Current (Big Brother)",
-            subpages: [{
-                name: "Contestants",
-                path: "/active/" + currentBBLeague + "/contestants"
-            }, {
-                name: "Scoring",
-                path: "/active/" + currentBBLeague + "/scoring"
-            }, {
-                name: "League Standing",
-                path: "/active/" + currentBBLeague + "/league-standing"
-            }]
+    // Necessary Node modules to fetch data
+    const fs = require('fs');
+    const path = require('path');
+    
+    // Based on availability in leagueConfiguration
+    const pathToLeagueData = path.join(process.cwd(), 'app', 'leagueConfiguration');
+    const activeLeaguePaths:Array<ILeagueLink> = [];
+    const archiveLeaguePaths:Array<ILeagueLink> = [];
+    fs.readdirSync(pathToLeagueData).map((file: string) => {
+      // Needed status for url
+      const { LEAGUE_STATUS } = require(`../leagueConfiguration/${file}`);
+      // Parses filename and converts it to url format
+      const showAndSeason = file.split('_');
+      const showNameArray = showAndSeason[0].split(/(?<![A-Z])(?=[A-Z])/);
+      const showNameFormatted = showNameArray.join('-').toLowerCase();
+      const showSeason = showAndSeason[1].replace('.js', '');
+      const showNameAndSeason = `${showNameFormatted}-${showSeason}`;
+      let friendlyName = `${showNameArray.join(' ')} ${showSeason}`;
```

Comment:
fft: I just wanna say, I really dislike all this parsing code. It is very dense and to have it in the middle of a much larger function whose purpose to to achieve something generally unrelated to this makes this section kind of jarring.